pyxem/tests/signals/test_polar_diffraction2d.py

```diff
@@ -142,6 +143,46 @@ def test_power_inplace(self, flat_pattern):
         assert ac is None
         assert isinstance(flat_pattern, Power2D)
 
+class TestPearsonCorrelation:
+    @pytest.fixture
+    def flat_pattern(self):
+        pd = PolarDiffraction2D(data=np.ones(shape=(2, 2, 5, 5)))
+        pd.axes_manager.signal_axes[0].scale = 0.5
+        pd.axes_manager.signal_axes[0].name = "theta"
+        pd.axes_manager.signal_axes[1].scale = 2
+        pd.axes_manager.signal_axes[1].name = "k"
+        return pd
+
+    @pytest.mark.parametrize("kmin", [0, 1])
+    @pytest.mark.parametrize("kmax", [4, 5])
+    def test_pcorrelation_signal(self, flat_pattern, kmin, kmax):
+        rho_w = flat_pattern.get_pearson_correlation()
+        assert isinstance(rho_w, Symmetry1D)
+        rho = flat_pattern.get_pearson_correlation(kmin=kmin, kmax=kmax)
+        assert isinstance(rho, Symmetry1D)
+
+    def test_pcorrelation_inplace(self, flat_pattern):
+        rho = flat_pattern.get_pearson_correlation(inplace=True)
+        assert rho is None
+        assert isinstance(flat_pattern, Symmetry1D)
+
+    def test_axes_transfer(self, flat_pattern):
+        rho = flat_pattern.get_pearson_correlation()
+        assert (
+            rho.axes_manager.signal_axes[0].scale
+            == flat_pattern.axes_manager.signal_axes[0].scale
+        )
+
+    @pytest.mark.parametrize(
+        "mask", [None, np.zeros(shape=(5, 5))]
+    )
+    def test_masking_pcorrelation(self, flat_pattern, mask):
+        rho_0 = flat_pattern.get_pearson_correlation(mask=mask)
+        print(rho_0)
```

Comment:
pytest will capture these print statements as far as I know, so we should delete them to keep the trace clean